packages/dht/src/connection/webrtc/BrowserWebrtcConnection.ts

```diff
@@ -179,10 +194,16 @@ export class NodeWebrtcConnection extends EventEmitter<Events> implements IWebrt
 
     public send(data: Uint8Array): void {
         if (this.lastState === 'connected') {
-            if (this.dataChannel!.bufferedAmount > this.bufferThresholdHigh) {
-                this.messageQueue.push(data)
-            } else {
-                this.dataChannel?.send(data as Buffer)
+        
+            this.dataChannel?.send(data as Buffer)
+            
+            let bufferedAmountRemainder = this.dataChannel!.bufferedAmount - data.length
+            bufferedAmountRemainder = Math.max(0, bufferedAmountRemainder)
+            if (bufferedAmountRemainder !== this.currentBufferedAmount) {
+                this.currentBufferedAmount = bufferedAmountRemainder
+                // Update bufferedAmount in statistics but keep the upload rate the same
+                this.statistics.bufferedAmount = this.currentBufferedAmount
+                this.emit('statisticsUpdated', this.statistics)
```

Comment:
Does this event end up being called quite often? Could there be some performance related if the bufferedAmount changes on each send

Reply:
The buffered amount should stay at 0 in normal operation, and be non-zero only if you are sending too fast. Maybe we should test this out with streamrtv by making a rc release of this branch?

Reply:
Okay, I suppose it would make sense to do an internal release of these changes